ttnn/cpp/ttnn/operations/pool/upsample/device/upsample_program_factory_multicore.cpp

```diff
@@ -55,84 +72,122 @@ static Tensor create_config_tensor(
         }
     }
 
-    std::vector<uint16_t> logical_core_to_stick_map;
-    const size_t logical_core_to_stick_map_entry_size = 3;
-    size_t row_size = logical_core_to_stick_map_entry_size * in_w;
+    std::vector<StickInterval> logical_core_to_stick_map;
+    std::vector<uint16_t> dst_core_end_idx_map;  // used for padding the config vector per reader
+
+    bool reader_sticks_reached = false;
+    bool insert_new_interval = false;
+    uint32_t last_ind = 0;
+    uint32_t elems_per_core_reader = 0;  // represents number of intervals per reader (NCRISC/BRISC)
+    uint32_t elem_num = 0;
     // Create map of core and respective offsets in input
     for (uint32_t b = 0; b < batch_size; ++b) {
         for (uint32_t h = 0; h < in_h; ++h) {
-            for (uint32_t w = 0; w < in_w; ++w, ++stick_offset) {
-                if (stick_offset == input_nsticks_per_core) {
-                    stick_offset = 0, ++core_idx;
+            for (uint32_t j = 0; j < scale_factor_h; ++j) {
+                stick_offset_start = stick_offset;
+                core_idx_start = core_idx;
+
+                for (uint32_t w = 0; w < in_w; ++w, ++stick_offset, ++stick_cnt) {
+                    if (stick_offset == input_nsticks_per_core) {
+                        stick_offset = 0;
+                        core_idx++;
+                    }
+                    reader_sticks_reached = stick_cnt == output_nsticks_per_core_reader;
+                    insert_new_interval = insert_new_interval || stick_offset == 0 || reader_sticks_reached;
+
+                    if (!insert_new_interval) {
+                        logical_core_to_stick_map.back().offset_end++;
+                        continue;
+                    }
+                    if (reader_sticks_reached) {
+                        dst_core_end_idx_map.push_back(logical_core_to_stick_map.size());
+                        if (output_nsticks_per_core > 1) {
+                            output_nsticks_per_core_reader = output_nsticks_per_core - output_nsticks_per_core_reader;
+                        }
+                        stick_cnt = 0;
+                        elem_num = logical_core_to_stick_map.size() - last_ind;
+                        elems_per_core_reader = elem_num > elems_per_core_reader ? elem_num : elems_per_core_reader;
+                        last_ind = logical_core_to_stick_map.size();
+                    }
+                    insert_new_interval = false;
+                    if (is_height_sharded) {
+                        logical_core_to_stick_map.push_back(
+                            StickInterval(logical_cores[core_idx].x, logical_cores[core_idx].y, stick_offset));
```

Comment:
you can use emplace_back (vs push_back) and in that case you remove explicit ctor